lib/test.js

```diff
@@ -225,29 +225,25 @@ Test.prototype._assert = function assert (ok, opts) {
             }
             
             var s = m[1].split(/\s+/);
-            var filem = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/.exec(s[1]);
-            if (!filem) {
-                filem = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/.exec(s[2]);
-                
-                if (!filem) {
-                    filem = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/.exec(s[3]);
-
-                    if (!filem) {
-                        continue;
-                    }
-                }
+            var filemRe = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/;
+            var filem;
+            var sIndex;
+            for (sIndex in s.slice(0, 4)) {
+                filem = filemRe.exec(s[sIndex]);
+                if (filem) break;
             }
+            if (! filem) continue;
             
             if (filem[1].slice(0, dir.length) === dir) {
                 continue;
             }
-            
-            res.functionName = s[0];
+
+            res.functionName = s.length > 1 ? s[0] : '<anonymous>';
             res.file = filem[1];
             res.line = Number(filem[2]);
             if (filem[3]) res.column = filem[3];
             
-            res.at = m[1];
+            res.at = s.length > 1 ? m[1] : '<anonymous> (' + m[1] + ')';
```

Comment:
why s.length instead of m.length here?

Reply:
`m` is the result of the regular expression matching something like `at /path/to/file.js:123:10` or `at functionName /path/to/file.js:123:10`. If we're at this point in the code, `m.length` should always be the same, where `m[1]` is the portion of the regex matching everything after `at`.

`s` is `m[1].split(/\s+/)` So if `s` has a length of 1, we know that`m[1]` is just case where there's just a path with no function name.

Although in writing this, it now occurs to me that the original `m[1].split(/\s+/)` code fails to account for directory and file names with spaces ... lemme go fix that.

Reply:
Maybe we could store `s.length > 1` in a nicely named variable, for clarity?